### ISSUE TITLE
`extensionResource` should be able to assign the resource type of the scope parameter

### ISSUE THREAD

**Issue by ArcturusZhang** [Azure/typespec-azure#1270]
We have a `@extensionResource` which allows us to construct a resource uri like:
```
/{resourceUri}/providers/namespace/do/something
```

but in real cases the `resourceUri` might need to specify resource types - some Rp's scope resource could only support some particular resource type.

Also the type of the parameter reduced from this decorator should be `armResourceIdentifier` - if we made this change, it has the ability to accept a resource type list for validate resource types.

**Comment by markcowl** [comment]
@ArcturusZhang This is frialy common for extension resources that only apply to VMs, but there are a few extension resource types that can apply to a few resources.  Do you think using the `allowedResourceTypes` in the armResourceIdentifier could be used to specify particular resource types

**Comment by ArcturusZhang** [comment]
> @ArcturusZhang This is fairly common for extension resources that only apply to VMs, but there are a few extension resource types that can apply to a few selected resources. Do you think using the `allowedResourceTypes` in the armResourceIdentifier could be used to specify particular resource types for this parameter?

yeah i am thinking the `allowedResourceTypes` should allow us to do that.

Also for extension resources that could only apply to one specific resource type they usually are defined very explicitly like this:
```
/subscriptions/{subsId}/resourceGroups/{rgName}/providers/Microsoft.Compute/virtualMachines/{vmName}/providers/Microsoft.GuestConfigurations/configurations/{name}
```
How do we write typespec for this case? A real case here: https://github.com/Azure/azure-rest-api-specs/blob/129c4f812edc27a0e4bd27225a1f7e296fadd160/specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/stable/2024-04-05/guestconfiguration.json#L30

**Comment by JoseLimaIVD** [comment]
My team would also desire to have this feature

**Comment by markcowl** [comment]
The ability to specify resource types was added in the last release.  Closing
